### ISSUE TITLE
Allow or force editors to provide contextual alt text for images used in rich text fields

### ISSUE THREAD

**Issue by skleinfeldt** [plone/Products.CMFPlone#2836]
When inserting an image into rich text using TinyMCE, the editor should be presented with 3 things:

1. A link to a popup containing information about alt text rules. The link should say "Rules for the proper use of alt text". Popup text is given below.
2. A plain text field for entering alt text: label is "Alt Text", help text is "Describe the meaning of the image for people using assistive technology like screen readers"
3. A plain text field for entering a comment: label is "Alt Text Not Required", help text is "If Alt Text is not required for this image, explain why (see link to rules given above)"

Note that for 3. the CastleCMS implementation is to save the comment with the content item (not the image). We should follow the Castle implementation; if there are multiple images with comments, they should be concatenated.

Either the Alt Text OR the Alt Text Not Required field can be filled out, but not both. If the accessibility control panel (#2834) is set to force editors to enter contextual alt text, either the alt text field or the comment field is required.

Popup text:

Alt (alternative) text should be provided for all images used in rich text so that people using assistive technology like screen readers can understand the meaning of the image in the context of the surrounding text. 

- If the image contains communicative text (not text included for visual effect), use the alt text to provide the text of the image.
- If the image is a simple graphic or photograph, use the alt text to briefly describe its meaning to the current page or context.
- if same image is embedded multiple times/places, it can have different alt text each place it is embedded.

There are a few exceptions to this rule:

- If the text in the image is also present as real text nearby, the alt text should be empty. 
-  If the image is a graph or complex piece of information, include the information contained in the image elsewhere on the page but leave the alt text empty.
- If the image is purely decorative or not intended for the user, the alt text should be empty.

**Comment by skleinfeldt** [comment]
More information about alt text: This blog post provides [an overview of alt text functionality](https://mrwweb.com/alt-text-decision-tree/) and a link to the W3C's [simple alt text decision tree](https://dev.w3.org/html5/alt-techniques/developer.html#tree).

**Comment by jensens** [comment]
Still relevant for Plone 6. Prompting editors for alt text when inserting images in the editor is a concrete a11y improvement. TinyMCE integration details would need updating for Plone 6.
